### PR TITLE
Grants the common man access to some parts of the lavaland mining base

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
## About The Pull Request

Checking out the diff render on mapdiffbot is going to give you a better picture, but basically this PR:
1. connects the gulag base to the mining base via a security office
2. gives gulag a scrubber and a vent from the mining base atmos
3. adds a new public shuttle to meta, box and delta that sits on the whiteship dock at roundstart.
4. replaces the storage room with the three extra mining closets and a borg charger no-one will ever use with a shuttle foyer containing gulag-tier mining shit for people who want to go out. Bring your own internals. 
5. splits the mining hallway and connects the lower half with the ore processing room to create a new area.
6. adds the items to make the shuttles work

The mapping is not the best but it should be functional and "good enough."

## Why It's Good For The Game

I believe allowing people to get into lavaland will better "integrate" the mining station into a standard round of ss13. Some jobs already have lavaland-content that is locked behind them handling the aux base stuff, like engineers and curators. Other jobs don't have roundstart lavaland access but have minor content there, like botanists. This PR will let those jobs take more advantage of the opportunities in lavaland, and also make better use out of the left side of the mining station that only really sees anything happen during a cult round, when one of the miners is a cultist. 

The plan isn't to make a planetstation or anything like that, but if in highpop a couple more people find something to do in lavaland then I'd say it's a win. 

## Changelog
:cl: Naksu
add: Mining base now has a common area accessible via a new shuttle on the medium-highpop maps, and a security office connecting the gulag and the mining base. Gulag also has functional atmos.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
mapdiffbot is dead can you upload to https://affectedarc07.co.uk/TGStuff/MapGen/?p=1

**Comment by vuonojenmustaturska** [comment]
image
here's an image from the dreammaker window, mapdiffbot seems to be disconnected from the tg repo

**Comment by HomocidlePancakes** [comment]
Gulag nerf

**Comment by Tlaltecuhtli** [comment]
does this make anyone with mining access able to free gulag prisoners?

**Comment by ATH1909** [comment]
> Gulag nerf

Oh hey, you're right, it IS gonna be harder to break out of the gulag now.

I think you can still do it if you throw enough objects at the reinforced window (and its grille) just south of the ore crates to break it, then perform the standard self-bat+extinguisher trick.

**Comment by vuonojenmustaturska** [comment]
it's not any harder to break out of gulag other than on the off-chance that a security guy is posted in lavaland

**Comment by HomocidlePancakes** [comment]
>Oh hey, you're right, it IS gonna be harder to break out of the gulag now.

Wut. I am suprised I have to explain how it’s EASIER to break out but mkay.

1)Higher chance to tele out with BS crystals.
2)Guranteed break out using low-tier gyntonate to break the walls.

**Comment by kriskog** [comment]
Probably a good change. The annoying people who wanna go lavaland hunting the miners' shit usually finds a way down there anyway.

**Comment by MegaEmpirical** [comment]
Walls around sec area should be reinforced at least.

**Comment by Dawson1917** [comment]
Please keep the spare equipment lockers somewhere on the base. Maybe the SMES room? They're very useful when you lost some gear or are one of the few people who use two PKAs at once. As well as anyone turned into a miner due to them all, frequently, dying immediately into the shift benefiting from spare lockers to grab a PKA because picks are terrible to mine with.

**Comment by FloranOtten** [comment]
Four things:

1. The sec room looks janky as fuck. You can tell that it's just an attempt to stitch two existing structures together. It's massive for the purpose it fulfills and it's really empty.
2. Escape proofing. It's worse than the permabrig. Sure, the lava river widening will make it harder to smash yourself across the lava or walk it, but there's some clever new ways to get out;
Gibtonite: Stabilize gibtonite. Destabilize it back at the sec office wall. Do this twice. You are now at mining base proper.
Helpful miners: Yeah, you could arrange for a miner to extinguish you across the lava, but now you don't even need to risk the damage. A miner can just hack the doors and you're free.
Doors: Bold of you to assume I won't stand there for a few minutes chipping away at those doors with a mining tool.
Hidden emag: Just use a storage implant and emag your way free!
3. The equipment lockers; Please keep them. I like them. Countless times have I used it for a spare protokinetic. 
4. Cult basing. I like this for its take on cult bases. Instead of being able to just knock out comms and the two consoles to prevent anyone from escaping Lavaland, then slaughtering anyone coming down with your massively overpowered gear, you now need to contend with two shuttles. One of these shuttles comes straight from Security. This shuttle may also be used to smash into security.

**Comment by vuonojenmustaturska** [comment]
I didn't widen the lava river though, it's been that way since mid-may.

**Comment by MegaEmpirical** [comment]
> Four things:
> 
> 1. The sec room looks janky as fuck. You can tell that it's just an attempt to stitch two existing structures together. It's massive for the purpose it fulfills and it's really empty.
> 2. Escape proofing. It's worse than the permabrig. Sure, the lava river widening will make it harder to smash yourself across the lava or walk it, but there's some clever new ways to get out;
>    Gibtonite: Stabilize gibtonite. Destabilize it back at the sec office wall. Do this twice. You are now at mining base proper.
>    Helpful miners: Yeah, you could arrange for a miner to extinguish you across the lava, but now you don't even need to risk the damage. A miner can just hack the doors and you're free.
>    Doors: Bold of you to assume I won't stand there for a few minutes chipping away at those doors with a mining tool.
>    Hidden emag: Just use a storage implant and emag your way free!
> 3. The equipment lockers; Please keep them. I like them. Countless times have I used it for a spare protokinetic.
> 4. Cult basing. I like this for its take on cult bases. Instead of being able to just knock out comms and the two consoles to prevent anyone from escaping Lavaland, then slaughtering anyone coming down with your massively overpowered gear, you now need to contend with two shuttles. One of these shuttles comes straight from Security. This shuttle may also be used to smash into security.

Some additional commentary on these points. 

1) Could improve space usage by double airlocking at least the gulag side if not both. 

2) Agreed, it is very easy to escape with this. Reinforced walls are needed at the least. 

3) Agreed.

4) Three shuttles it seems, so this is quite nice.

**Comment by ArcaneMusic** [comment]
Couple Questions and Notes:
* Is the security locker on base a full security locker or a checkpoint style locker (Does it have a security belt)?

* Would it be possible to push the  lobby to the southwest a few tiles thinner to keep the supply closet with the extra mining gear?

* It may be necessary to add another oxygen tank to the civilian side of the base as you're now going to expand the air supply to comfortably house more people, but I would only see this being necessary on rounds like meteor or disaster if it ever comes back, so that could remain the same no problem.

* That being said in a roundtype like disaster, this base would probably suffer FAR less than the old base used to, considering that people being shitters in such a small volume meant you didn't really have a reliable way to arrest someone if they WERE being a shit, now it really does function like a miniature isolated station.

*Now, if ashlizards are attacking the mining station, the miners can ask for assistance from security, and likewise, the ashlizards can now break the security vendor for access to one or two pieces of security equipment. 

Overall, I think is a solid addition.

**Comment by vuonojenmustaturska** [comment]
The security locker is a /cargo subtype, whatever that means. 

While I COULD put the storage room back in, I'm definitely against it as the extras miners can take from the storage are items they can get via the voucher and via the vending machine. The lockers never made much sense gameplay design-wise as they just pretty much remove the need of ever buying more of these items or choosing more of them as your voucher option.

**Comment by MegaEmpirical** [comment]
They're there for when the HoP adds more miners as they often die and shit. Or if they die and get cloned and need new shit.

**Comment by Dawson1917** [comment]
> 
> 
> The security locker is a /cargo subtype, whatever that means.

Wouldn't that be the same as the cargo sec office?
If so then it'll have 1x seclite, 1x cargo armband, 1x supply radio encryption key, 1x helmet, 1x armor vest, **1x secHUDs, 1x security bowman headset, and 1x security headset.** Different from the regular sec checkpoint locker at arrivals in that it has the cargo armband and radio key. I don't think miners should have an easy way to loot a free sechud and sec bowman in an area nobody will likely ever notice has been looted. God knows it's easy enough to loot the sec locker in cargo, and that at least gets noticed either by sec, the AI, or a snitch in cargo.

**Comment by YakumoChen** [comment]
Not a fan of making it easier to escape gulag or giving mining essentially free security loot. This makes it almost not worth sending people down at all since it'll be piss easy to break out, and most people sent down there are shitters who will break out any chance they get.

**Comment by dr-bee** [comment]
Double stack reinforced walls on the sec checkpoint to prevent gibonite cheese. This still makes it possible to emag out of the gulag with a storage implant and emag but that is a 14TC problem solution so honestly it shouldnt be too much of a nerf. 

I love this change for engineering, It would let me do fun shit with lava and TEG engines easier.

**Comment by nero1024** [comment]
Why doesn't anyone just ask the HoP for lavaland/mining access? I mean Lavaland is technically in the HoPs domain too along with QM.

**Comment by MegaEmpirical** [comment]
The emag issue can be resolved with alternating airlocks that bolt on close like in virology.

**Comment by CPTANT** [comment]
Nobody will escape the gulag after this because sec is never going to bother putting a prisoner in such an easily escaped prison in the first place.

**Comment by FloranOtten** [comment]
Also, won't the default docked shuttle fuck with the whiteship? Currently, if that dock is occupied (as with the Fugitive ship) the Whiteship refuses to dock there. Maybe add a new shuttle dock for it? Or have it dock at the transit pod dock?

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -1,90 +1,98 @@
...
-		return
-	. = ..()
-
-/**********************Mining car (Crate like thing, not the rail car)**************************/
-
-/obj/structure/closet/crate/miningcar
-	desc = "A mining car. This one doesn't work on rails, but has to be dragged."
-	name = "Mining car (not for rails)"
-	icon_state = "miningcar"
+/**********************Light************************/
```
Comment (ninjanomnom) [inline code/modules/mining/mine_items.dm]: Line ending fuckup?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by MacBlaze1** [comment]
Good change, but I recommend Reinforced walls all around, maybe increase the liveliness of the sec guard room and make it smaller, if we're going to be extra careful then vault doors with wire protection or something like the abandoned space base with 2 heavy blast doors for a room that transfers prisoners to certain places.

**Comment by vuonojenmustaturska** [comment]
@FloranOtten it might, but i don't want to make any extensive mapping in the station maps until smart cables etc are done for good. 

The white ship is such a minor feature that I don't expect any real problems, it has a custom docker anyways

**Comment by vuonojenmustaturska** [comment]
image

File: code/modules/mining/mine_items.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/modules/mining/mine_items.dm]: the original file probably had LF line endings

**Comment by Mey-Ha-Zah** [comment]
I see several huge problem with this.

1: A Miner antag could free gulag prisoners, and depending on the mode he can convert/kill or whatever he wants with them.

2: The reason mining base is separate from the station is its supposed to be a mining base, not a relaxation bar. People can convert the aux base into a space bar if they want to main base is not supposed to be a public zone.

3: White ship dock. Hell no son. You need to make a new dock, not just shunt a feature aside because you believe yours is better. Find somewhere else to park its ass. There is plenty of airlocks in maint.

4: Mining base, in my opinion, actually needs to be shrunken down, not expanded. Its already huge for its VERY minor needs.

**Comment by YakumoChen** [comment]
Doesn't this just entirely make the aux base pointless? The Aux base is supposed to be a more public version that you can send down wherever on the mine.

**Comment by vuonojenmustaturska** [comment]
Miners already could free gulag prisoners, this PR doesn't change that. On the flipside, station crew with kill objectives on miners can now do better than hope the miner runs into a chasm.

AFAIK the reason why the mining base is separate from the station is that it's a placeholder for a hub into exciting bossrush singleplayer content that never happened as designed and I don't care about. As for needs, the only things people use on the mining base are the ore boxes, the suit lockers, and the spare lockers in storage for extra bluespess capsules. You could fill the miners lavaland-side needs with a shuttle console on always-powered turf and remove the whole mining base if you moved the suit lockers stationside. 

But the point is, the mining base already exists, if we make use of it we can expand gameplay options for the crew for free. It'll even bring mining closer to being a crew position just by increasing the interaction surface between them and the station crew. 

I'm using the white ship dock because it's there, it's accessible to the crew, it's borderline useless and using it doesn't require any extensive remapping of the station that would possibly cause issues with bigger PRs like smart cables. It's not intended to be a long-term solution.

As for the aux base, there are similar goals here and I did want to use the aux base dock but the aux base offers gameplay my PR doesn't. The aux base will definitely see less use now since people using it to go down no longer need to take a gamble on where they'll land.

**Comment by Mey-Ha-Zah** [comment]
Your answers don't clear up the issues. And claiming something I borderline useless is arrogant. I for one spent several rounds always getting the white ship and turning it into my command ship. Admins sometimes use it for space missions, and by re-assigning the dock to your shuttle, you completely remove that option. To be honest, I've played some rounds, and by giving people access to the mining station, all you've done is allowed the greytide to swarm there as well, grab weapons and tools meant for mining and just generally be a nuisance.

Also, targets being hard to kill for traitors is part of the challenge. You can't expect to be given a quick route to your target for every assassination type. Should there be a public door into security so players don't have to hack their way in so they can scope out killing the HOS?

And finally as former miner main, and one of the main spriters behind a potential lavaland rework, I'm gonna have to say this PR is not an improvement, and would make my job remapping the Mining base into the more compact version I have planned.

**Comment by Mey-Ha-Zah** [comment]
TLDR: Lavaland isn't a vacation spot, it's a deadly mining site with ancient and powerful relics. The rest of the station has no need to be there, so why should you open it up to them?

**Comment by CPTANT** [comment]
> TLDR: Lavaland isn't a vacation spot, it's a deadly mining site with ancient and powerful relics. The rest of the station has no need to be there, so why should you open it up to them?

Because it's fun and the mining station is way too isolated.

**Comment by vuonojenmustaturska** [comment]
Admins can move shuttles, even ACs are taught how to. 

As for your lavaland work, I will give deference to existing work for minor parts of my PR like here where I'm avoiding major touches on the big three stations while WJ and actioninja do smart cables. This is an effort well underway whereas you have been talking about your mining rework since sheridan was headcoder, and I recall your plan being "remove ore collection from mining, add more Z-levels containing more single-player content" which is something I fundamentally disagree with.

I want mining to be a "station role" even if it happens on a separate Z-level, rather than being a quasi- ghost-role that doesn't really interact with the station.

**Comment by vuonojenmustaturska** [comment]
Actually, I may have confused you with @MrDoomBringer

**Comment by py01** [comment]
Fauna can spawn inside the mining base it seems
685f894272406cb3061080a3155e741f

**Comment by vuonojenmustaturska** [comment]
that's a statue

File: code/game/objects/structures/crates_lockers/closets/secure/freezer.dm [hunk 2]
```diff
@@ -49,6 +49,13 @@
 	req_access = null
 	locked = FALSE
 
+/obj/structure/closet/secure_closet/freezer/gulag_fridge
+	name = "refrigerator"
+
+/obj/structure/closet/secure_closet/freezer/gulag_fridge/PopulateContents()
+	..()
+	for(var/i = 0, i < 3, i++)
```
Comment (ninjanomnom) [inline code/game/objects/structures/crates_lockers/closets/secure/freezer.dm]:
```suggestion
	for(var/i in 1 to 3)
```

File: code/datums/shuttles.dm [hunk 3]
```diff
@@ -124,6 +124,13 @@
 	port_id = "mining"
 	can_be_bought = FALSE
 
+/datum/map_template/shuttle/mining_common
+	port_id = "mining_common"
+	can_be_bought = FALSE
+
+/datum/map_template/shuttle/mining_common/meta
+	suffix = "meta"
```
Comment (ninjanomnom) [inline code/datums/shuttles.dm]: This is the base shuttle types section, move the actual shuttle to the section below with the rest of the actual shuttles

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: _maps/shuttles/mining_common_meta.dmm [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	name = "Lavaland Shuttle Airlock"
+	},
+/obj/docking_port/mobile{
+	dir = 8;
+	dwidth = 3;
+	height = 5;
+	id = "mining_common";
+	name = "lavaland shuttle";
+	port_direction = 4;
+	width = 7
```
[the remark this answers is not in the document]
Comment (vuonojenmustaturska) [reply _maps/shuttles/mining_common_meta.dmm]: as mentioned in discord, doing that makes the shuttle only load the objects that are on the docking port turf for whatever reason.

[the remark this answers is not in the document]
Comment (ninjanomnom) [reply _maps/shuttles/mining_common_meta.dmm]: It's probably not moving things after load properly. I'll take a look at this at a later date since it doesnt seem to be a bug in this pr

**Review by ninjanomnom — APPROVED** [review]

**Comment by Shadowflame909** [comment]
This is basically the aux base again

we got two aux bases now
